src/core/hle/service/boss/boss.cpp

```diff
@@ -17,6 +37,118 @@ void Module::Interface::InitializeSession(Kernel::HLERequestContext& ctx) {
+
+    // Read the file if it already exists
+    if (boss_a_result.Succeeded()) {
+        auto boss_a = std::move(boss_a_result).Unwrap();
+        if (boss_a->GetSize() > boss_save_header_size &&
+            ((boss_a->GetSize() - boss_save_header_size) % boss_a_entry_size) == 0) {
+            u64 num_entries = (boss_a->GetSize() - boss_save_header_size) / boss_a_entry_size;
+            for (u64 i = 0; i < num_entries; i++) {
+                u64 prog_id;
+                boss_a->Read(i * boss_a_entry_size + boss_save_header_size, sizeof(prog_id),
+                             (u8*)&prog_id);
+                LOG_DEBUG(Service_BOSS, "id in entry {} is {:#018X}", i, prog_id);
+            }
+        }
+    }
+    FileSys::Path boss_sv_path("/BOSS_SV.db");
+
+    auto boss_sv_result = boss_system_save_data_archive->OpenFile(boss_sv_path, open_mode);
+
+    FileSys::Path boss_ss_path("/BOSS_SS.db");
+
+    auto boss_ss_result = boss_system_save_data_archive->OpenFile(boss_ss_path, open_mode);
+
+    // Read the files if they already exists
+    if (boss_sv_result.Succeeded() && boss_ss_result.Succeeded()) {
+        auto boss_sv = std::move(boss_sv_result).Unwrap();
+        auto boss_ss = std::move(boss_ss_result).Unwrap();
+        if (boss_sv->GetSize() > boss_save_header_size &&
+            ((boss_sv->GetSize() - boss_save_header_size) % boss_s_entry_size) == 0 &&
+            boss_sv->GetSize() == boss_ss->GetSize()) {
+            u64 num_entries = (boss_sv->GetSize() - boss_save_header_size) / boss_s_entry_size;
+            for (u64 i = 0; i < num_entries; i++) {
+                u64 prog_id;
+                const u64 prog_id_offset = 0x10;
+                boss_sv->Read(i * boss_s_entry_size + boss_save_header_size + prog_id_offset,
+                              sizeof(prog_id), (u8*)&prog_id);
+                LOG_DEBUG(Service_BOSS, "id sv in entry {} is {:#018X}", i, prog_id);
+                std::string task_id(task_id_size, 0);
+                const u64 task_id_offset = 0x18;
+                boss_sv->Read(i * boss_s_entry_size + boss_save_header_size + task_id_offset,
+                              task_id_size, (u8*)task_id.data());
+                size_t task_id_len = strnlen(task_id.c_str(), task_id_size);
+                task_id.resize(task_id_len < task_id_size ? task_id_len + 1 : task_id_size);
+                LOG_DEBUG(Service_BOSS, "task id in entry {} is {}", i,
+                          std::string(task_id, 0, task_id.size() - 1));
+                std::string url(sizeof(cur_props.x7), 0);
+                const u64 url_offset = 0x21C;
+                boss_ss->Read(i * boss_s_entry_size + boss_save_header_size + url_offset,
+                              sizeof(cur_props.x7), (u8*)url.data());
+                size_t url_len = strnlen(url.c_str(), url.size());
+                url.resize(url_len < url.size() ? url_len + 1 : url.size());
```

Comment:
strnlen(url.c_str(), url.size());
```suggestion
                size_t url_len = strnlen(url.c_str(), url.size()-1);
                url.resize(url_len + 1);
```